lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -454,25 +456,33 @@ func (h *StorageHandlersImpl) VolumeJoin(params storage.VolumeJoinParams) middle
 	//Note: Name should already be populated by now.
 	volume, err := h.volumeCache.VolumeGet(op, params.Name)
 	if err != nil {
-		log.Errorf("Volumes: StorageHandler : %#v", err)
+		op.Errorf("Volumes: StorageHandler : %#v", err)
 
 		return storage.NewVolumeJoinInternalServerError().WithPayload(&models.Error{
 			Code:    http.StatusInternalServerError,
 			Message: err.Error(),
 		})
 	}
 
-	actualHandle, err = vsphere.VolumeJoin(op, actualHandle, volume, params.JoinArgs.MountPath, params.JoinArgs.Flags)
+	switch volume.Device.DiskPath().Scheme {
```

Comment:
Can we move this into the volume cache?  Above we do a `VolumeGet`.  We should be able to do an `h.volumeCache.VolumeJoin` on the ID used to do the get.   The cache can then do a lookup on the ID, find the volumestorer impementation, and call volume join on the right implementation.

I feel like that's much cleaner than we we have here.  I don't want to leak details about nfs or vsphere in the swagger handler like we're doing here.  If you make a new issue, we can get to it next sprint.